src/main/java/com/atherys/towns/listener/ProtectionListener.java

```diff
@@ -1,58 +1,106 @@
 package com.atherys.towns.listener;
 
+import com.atherys.towns.api.permission.world.WorldPermissions;
 import com.atherys.towns.facade.PlotFacade;
+import com.atherys.towns.facade.ProtectionFacade;
 import com.google.inject.Inject;
 import org.spongepowered.api.block.BlockType;
-import org.spongepowered.api.block.BlockTypes;
 import org.spongepowered.api.entity.living.player.Player;
 import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.action.FishingEvent;
 import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.event.block.CollideBlockEvent;
 import org.spongepowered.api.event.block.InteractBlockEvent;
+import org.spongepowered.api.event.cause.entity.damage.source.IndirectEntityDamageSource;
+import org.spongepowered.api.event.entity.DamageEntityEvent;
 import org.spongepowered.api.event.entity.InteractEntityEvent;
+import org.spongepowered.api.event.entity.SpawnEntityEvent;
+import org.spongepowered.api.event.entity.projectile.LaunchProjectileEvent;
+import org.spongepowered.api.event.filter.cause.First;
 import org.spongepowered.api.event.filter.cause.Root;
-import org.spongepowered.api.event.item.inventory.InteractInventoryEvent;
+import org.spongepowered.api.event.item.inventory.UseItemStackEvent;
 
 public class ProtectionListener {
 
     @Inject
     PlotFacade plotFacade;
 
+    @Inject
+    ProtectionFacade protectionFacade;
+
     @Listener
     public void onBlockPlace(ChangeBlockEvent.Place event, @Root Player player) {
-        plotFacade.plotAccessCheck(event, player, true);
+        plotFacade.plotAccessCheck(event, player, WorldPermissions.BUILD, true);
     }
 
     @Listener
     public void onBlockBreak(ChangeBlockEvent.Break event, @Root Player player) {
-        plotFacade.plotAccessCheck(event, player, true);
+        plotFacade.plotAccessCheck(event, player, WorldPermissions.DESTROY, true);
     }
 
     @Listener
-    public void onBlockModify(ChangeBlockEvent.Modify event, @Root Player player) {
-        plotFacade.plotAccessCheck(event, player, true);
+    public void onProjectileLaunch(LaunchProjectileEvent event, @Root Player player) {
```

Comment:
Have you tested this? I haven't had good results with that event.

Reply:
This seemed to work in my testing for blocking any projectile-based event. What issues did you have? maybe I can test for those.

Reply:
Hmm. I don't really remember. Maybe it was fixed.

Reply:
Tested more projectiles to make sure, it seems to work fine at the moment.